rust/signer/src/util.rs

```diff
@@ -239,15 +284,64 @@ impl<'jni, Inner: Return<'jni, Env = JNIEnv<'jni>> + Default> Return<'jni> for O
                 // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                 let ret = Return::convert(env, Inner::default());
 
-                let class = env.find_class("java/lang/Exception").expect("Must have the Exception class; qed");
-                let exception: JThrowable<'jni> = env.new_object(class, "()V", &[]).expect("Must be able to instantiate the Exception; qed").into();
+                let class = env
+                    .find_class("java/lang/Exception")
+                    .expect("Must have the Exception class; qed");
+                let exception: JThrowable<'jni> = env
+                    .new_object(class, "()V", &[])
+                    .expect("Must be able to instantiate the Exception; qed")
+                    .into();
+
+                // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+                // !!!!                                                        !!!!
+                // !!!! WE CAN NO LONGER INTERACT WITH JNIENV AFTER THIS POINT !!!!
+                // !!!!                                                        !!!!
+                // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+                env.throw(exception)
```

Comment:
These error handling part seems duplicate with the following one, is it possible to wrap them into a function?

Reply:
It's not as straightforward as you might think, so I won't do it now.  Over time though I will absorb the Return/Arg traits into the `secure-native` lib, which is tidier, and remove them from this file.